### PR TITLE
Add usage and examples into Readme

### LINKED ISSUE #2: ToDo: Readme
To not forget.

### PR DESCRIPTION
**Pull request by AlexKVal**
Closes #2

Corrections / additions are welcome :cherries:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,5 +2,248 @@
...
+        return new Error('The block property requires the vertical property to be set to have any effect');
+      }
+    }
+  ])
+```
+
+All validators provided in an array as argument into `all()` will be validating
+property value each one by one.
+
+`all()` validator will success only when `all` provided validators success.
```
Comment (mtscout6) [inline README.md]: success => succeed

File: README.md [hunk 2]
```diff
@@ -2,5 +2,248 @@
...
+    React.PropTypes.bool,
+    function(props, propName, componentName) {
+      if (props.block && !props.vertical) {
+        return new Error('The block property requires the vertical property to be set to have any effect');
+      }
+    }
+  ])
+```
+
+All validators provided in an array as argument into `all()` will be validating
```
Comment (mtscout6) [inline README.md]: Perhaps: All validators will be validated one by one, stopping on the first failure.

File: README.md [hunk 3]
```diff
@@ -2,5 +2,248 @@
...
+property value each one by one.
```
Comment (mtscout6) [inline README.md]: Perhaps: All validators will be validated one by one, stopping on the first failure.

**Comment by AlexKVal** [comment]
@mtscout6 thank you for the help :cherries: 
Both are addressed.

File: README.md [hunk 4]
```diff
@@ -2,5 +2,247 @@
 
 [![Travis Build Status][build-badge]][build]
 
+This is a library of some custom validators for React components properties.
+Initially they were part of [React-Bootstrap](https://github.com/react-bootstrap/react-bootstrap/) project.
```
Comment (mathieumg) [inline README.md]: of the

File: README.md [hunk 5]
```diff
@@ -2,5 +2,247 @@
...
+```
+or
+```js
+import CustomPropTypes from 'react-prop-types';
+// and then used as usual
+propTypes: {
+  someProp: CustomPropTypes.elementType
+```
+
+If you use `webpack`, you can bundle only those validators you need like this:
```
Comment (mathieumg) [inline README.md]: If you use `webpack` and only want to bundle the validators you need, prefer the following approach:

File: README.md [hunk 6]
```diff
@@ -2,5 +2,247 @@
...
+    function(props, propName, componentName) {
+      if (props.block && !props.vertical) {
+        return new Error('The block property requires the vertical property to be set to have any effect');
+      }
+    }
+  ])
+```
+
+All validators will be validated one by one, stopping on the first failure.
+
```
Comment (mathieumg) [inline README.md]: The `all()` validator will only succeed when all validators provided also succeed.

File: README.md [hunk 7]
```diff
@@ -2,5 +2,247 @@
...
+```js
+import elementType from 'react-prop-types/elementType'
+```
+
+---
+#### all(arrayOfValidators)
+
+This validator allows to have complex validation like this:
+```js
+propTypes: {
```
Comment (mathieumg) [inline README.md:37]: Perhaps the example can be simplified a bit and become react-bootstrap agnostic? That might be stuff for a later PR however.

File: README.md [hunk 8]
```diff
@@ -2,5 +2,247 @@
...
+      if (props.block && !props.vertical) {
+        return new Error('The block property requires the vertical property to be set to have any effect');
+      }
+    }
+  ])
+```
+
+All validators will be validated one by one, stopping on the first failure.
+
+`all()` validator will succeed only when `all` provided validators succeed.
```
Comment (mathieumg) [inline README.md]: The `all()` validator will only succeed when all validators provided also succeed.

File: README.md [hunk 9]
```diff
@@ -2,5 +2,247 @@
...
+The type of element can be provided in two forms:
+- tag name (string)
+- a return value of `React.createClass(...)`
+
+Example
+```js
+propTypes: {
+  componentClass: CustomPropTypes.elementType
+```
+Then `componentClass` can be set like
```
Comment (mathieumg) [inline README.md]: Then, `componentClass` can be set by doing:

**Comment by AlexKVal** [comment]
@mathieumg thank you for the help :cherries: 
All remarks has been addressed.

``` diff
-Initially they were part of [React-Bootstrap](https://github.com/react-bootstrap/react-bootstrap/) project.
+Initially they were part of the [React-Bootstrap](https://github.com/react-bootstrap/react-bootstrap/) project.

-If you use `webpack`, you can bundle only those validators you need like this:
+If you use `webpack` and only want to bundle the validators you need, prefer the following approach:

-`all()` validator will succeed only when `all` provided validators succeed.
+The `all()` validator will only succeed when all validators provided also succeed.

-Then `componentClass` can be set like
+Then, `componentClass` can be set by doing:
```

> Perhaps the example can be simplified a bit and become react-bootstrap agnostic?

I could not devise a better example. The lack of creativity (and perhaps a bit of offline job these days).

**Comment by AlexKVal** [comment]
Need one more approval from a member of @react-bootstrap/collaborators

File: README.md [hunk 10]
```diff
@@ -2,5 +2,247 @@
...
+      </div>
+    );
+  }
+});
+
+React.render(<Example/>, mountNode);
+```
+
+---
+#### siglePropFrom(arrayOfPropertiesNames)
```
Comment (dozoisch) [inline README.md]: missing a `n` => single

**Comment by dozoisch** [comment]
LGTM except for that little `n` missing :+1:

**Comment by AlexKVal** [comment]
@dozoisch has `eagle eye` :eyes: :smile:

**Pull request merged**